File: backend/masz/Controllers/api/v1/Views/UserNoteViewController.ts
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.IO;
using System.Linq;
using System.Text.RegularExpressions;
using System.Threading.Tasks;
using masz.data;
using masz.Dtos.DiscordAPIResponses;
using masz.Dtos.ModCase;
using masz.Models;
using masz.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.JsonPatch;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.Logging;
using Microsoft.Extensions.Options;
using Newtonsoft.Json;

namespace masz.Controllers
{
    [ApiController]
    [Route("api/v1/guilds/{guildid}/usernoteview")]
    [Authorize]
    public class UserNoteViewController : SimpleController
    {
        private readonly ILogger<UserNoteViewController> logger;

        public UserNoteViewController(ILogger<UserNoteViewController> logger, IServiceProvider serviceProvider) : base(serviceProvider)
        {
            this.logger = logger;
        }

        [HttpGet]
        public async Task<IActionResult> GetGuildUserNoteView([FromRoute] string guildid)
        {
            logger.LogInformation($"{HttpContext.Request.Method} {HttpContext.Request.Path} | Incoming request.");
            if (! await this.HasPermissionOnGuild(DiscordPermission.Moderator, guildid)) {
                logger.LogInformation($"{HttpContext.Request.Method} {HttpContext.Request.Path} | 401 Unauthorized.");
                return Unauthorized();
            }

            List<UserNote> userNotes = await this.database.GetUserNotesByGuildId(guildid);
            List<UserNoteView> userNoteViews = new List<UserNoteView>();
            foreach (UserNote userNote in userNotes)
            {
                userNoteViews.Add(new UserNoteView() {
                    UserNote = userNote,
                    Moderator = await discord.FetchUserInfo(userNote.CreatorId, CacheBehavior.OnlyCache),
                    User = await discord.FetchUserInfo(userNote.UserId, CacheBehavior.OnlyCache)
                });
            }

            logger.LogInformation($"{HttpContext.Request.Method} {HttpContext.Request.Path} | 200 Returning list.");
            return Ok(userNoteViews);
        }
    }
}

File: backend/MASZ/Controllers/api/v1/Views/UserNoteViewController.cs
using System.ComponentModel.DataAnnotations;
using MASZ.Enums;
using MASZ.Models;
using MASZ.Repositories;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace MASZ.Controllers
{
    [ApiController]
    [Route("api/v1/guilds/{guildId}/usernoteview")]
    [Authorize]
    public class UserNoteViewController : SimpleController
    {
        public UserNoteViewController(IServiceProvider serviceProvider) : base(serviceProvider)
        {
        }

        [HttpGet]
        public async Task<IActionResult> GetGuildUserNoteView([FromRoute] ulong guildId, [FromQuery][Range(0, int.MaxValue)] int startPage = 0)
        {
            await RequirePermission(guildId, DiscordPermission.Moderator);

            List<UserNote> userNotes = await UserNoteRepository.CreateDefault(_serviceProvider, await GetIdentity()).GetUserNotesByGuild(guildId);

            int fullSize = userNotes.Count;

            List<UserNoteExpandedView> userNoteViews = new();
            foreach (UserNote userNote in userNotes.Skip(startPage * 20).Take(20))
            {
                userNoteViews.Add(new UserNoteExpandedView(
                    userNote,
                    await _discordAPI.FetchUserInfo(userNote.UserId, CacheBehavior.OnlyCache),
                    await _discordAPI.FetchUserInfo(userNote.CreatorId, CacheBehavior.OnlyCache)));
            }

            return Ok(new
            {
                items = userNoteViews,
                fullSize
            });
        }
    }
}